augur/filter/_run.py

```diff
@@ -381,23 +382,27 @@ def run(args):
             dropped_samps = list(sequence_strains - valid_strains)
             write_vcf(args.sequences, args.output, dropped_samps)
     elif args.sequences:
-        sequences = read_sequences(args.sequences)
-
         # If the user requested sequence output, stream to disk all sequences
         # that passed all filters to avoid reading sequences into memory first.
-        # Even if we aren't emitting sequences, we track the observed strain
-        # names in the sequence file as part of the single pass to allow
-        # comparison with the provided sequence index.
-        if args.output:
-            observed_sequence_strains = set()
-            with open_file(args.output, "wt") as output_handle:
-                for sequence in sequences:
-                    observed_sequence_strains.add(sequence.id)
-
+        # Even if we aren't emitting sequences, we check for duplicates and
+        # track the observed strain names in the sequence file as part of the
+        # single pass to allow comparison with the provided sequence index.
+        observed_sequence_strains = set()
+        duplicates = set()
+        with open_file(args.output, "wt") if args.output else nullcontext() as output_handle:
+            for sequence in read_sequences(args.sequences):
+                if sequence.id in observed_sequence_strains:
+                    duplicates.add(sequence.id)
+
+                observed_sequence_strains.add(sequence.id)
+
+                if args.output:
                     if sequence.id in valid_strains:
                         write_sequences(sequence, output_handle, 'fasta')
```

Comment:
just looking at the overall logic here (and without reading any of the other source so maybe i'm misunderstanding something), i wonder if the `if sequence.id in valid_strains` check should applied to the duplicate strain id detection too? (In other words, if there are duplicate sequence IDs, but they're not part of `valid_strains`, do we care?)

Reply:
Fair enough!